windows/devices/uwb/UwbDeviceConnector.cxx

```diff
@@ -395,46 +395,45 @@ UwbDeviceConnector::SetApplicationConfigurationParameters(uint32_t sessionId, st
-    //         // TODO: Could move this logic into its own UwbCxDdi::To() function
-    //         std::vector<std::tuple<::uwb::protocol::fira::UwbApplicationConfigurationParameterType, ::uwb::protocol::fira::UwbStatus>> appConfigParamsStatusList;
-    //         for (auto i = 0; i < uwbSetAppConfigParamsStatus.appConfigParamsCount; i++)
-    //         {
-    //             auto paramType = UwbCxDdi::To(uwbSetAppConfigParamsStatus.appConfigParamsStatus[i].paramType);
-    //             auto paramStatus = UwbCxDdi::To(uwbSetAppConfigParamsStatus.appConfigParamsStatus[i].status);
-
-    //             auto appConfigParamStatus = std::make_tuple(paramType, paramStatus);
-    //             appConfigParamsStatusList.push_back(appConfigParamStatus);
-    //         }
-    //         auto result = std::make_tuple(uwbStatus, std::move(appConfigParamsStatusList));
-    //         resultPromise.set_value(std::move(result));
-    //     }
-    // }
+
+    wil::shared_hfile handleDriver;
+    auto hr = OpenDriverHandle(handleDriver, m_deviceName.c_str());
+    if (FAILED(hr)) {
+        PLOG_ERROR << "failed to obtain driver handle for " << m_deviceName << ", hr=" << std::showbase << std::hex << hr;
+        resultPromise.set_exception(std::make_exception_ptr(UwbException(UwbStatusGeneric::Rejected)));
+        return resultFuture;
+    }
+
+    auto paramsBuffer = UwbCxDdi::From(applicationConfigurationParameters).data();
+    auto statusSize = offsetof(UWB_SET_APP_CONFIG_PARAMS_STATUS, appConfigParamsStatus[applicationConfigurationParameters.size()]);
+    std::vector<uint8_t> statusBuffer(statusSize);
+    BOOL ioResult = DeviceIoControl(handleDriver.get(), IOCTL_UWB_SET_APP_CONFIG_PARAMS, std::data(paramsBuffer), std::size(paramsBuffer), std::data(statusBuffer), statusSize, nullptr, nullptr);
+    if (!LOG_IF_WIN32_BOOL_FALSE(ioResult)) {
+        HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
+        PLOG_ERROR << "error when sending IOCTL_UWB_SET_APP_CONFIG_PARAMS with sessionId " << sessionId << ", hr = " << std::showbase << std::hex << hr;
+        resultPromise.set_exception(std::make_exception_ptr(UwbException(UwbStatusGeneric::Rejected)));
+        return resultFuture;
+    } else {
+        PLOG_DEBUG << "IOCTL_UWB_SET_APP_CONFIG_PARAMS succeeded";
+        auto& uwbSetAppConfigParamsStatus = *reinterpret_cast<UWB_SET_APP_CONFIG_PARAMS_STATUS*>(std::data(statusBuffer));
+        auto uwbStatus = UwbCxDdi::To(uwbSetAppConfigParamsStatus.status);
+        if (!IsUwbStatusOk(uwbStatus)) {
+            resultPromise.set_exception(std::make_exception_ptr(UwbException(std::move(uwbStatus))));
+        } else {
+            // TODO: Could move this logic into its own UwbCxDdi::To() function
+            std::vector<std::tuple<::uwb::protocol::fira::UwbApplicationConfigurationParameterType, ::uwb::protocol::fira::UwbStatus>> appConfigParamsStatusList;
+            for (auto i = 0; i < uwbSetAppConfigParamsStatus.appConfigParamsCount; i++)

```

Comment:
cpp linter says that this file doesn't comply with the style guideline, could you run the auto-format on this file?

Reply:
Admittedly I'm not sure how to do that, but I did make a minor formatting change that *might* fix it. How do you run auto-format?

Reply:
Looks like my change did fix it, but knowing how to run auto-format would still be good info

Reply:
You can use anything that invokes `clang-format` since we have a config file in the project. If you're using VSCode, it has the support built in: `Ctrl+P-> Format Document` will apply clang-format changes to the current file:
https://code.visualstudio.com/docs/editor/codebasics#_formatting

VS also supports clang-format and there's an option somewhere to enable it automatically:
https://devblogs.microsoft.com/cppblog/clangformat-support-in-visual-studio-2017-15-7-preview-1/

The `cpp-linter` PR check runs clang-format and adds comments for any cases that are non-compliant with the `.clang-format` file. We don't have it as a required check because it's sometimes too picky and the GitHub action has some bugs, but it's strongly preferred to have all new code be `clang-format` clean (hence the option in the PR checklist):
https://github.com/microsoft/nearobject-framework#coding-guidelines